pdns/pdns_recursor.cc

```diff
@@ -998,7 +1047,7 @@ void startDoResolve(void *p)
     if (luaconfsLocal->protobufServer) {
       pbMessage.setBytes(packet.size());
       pbMessage.setResponseCode(pw.getHeader()->rcode);
-      pbMessage.setAppliedPolicy(appliedPolicy);
+      pbMessage.setAppliedPolicy(appliedPolicy.d_name);
```

Comment:
Since your `preresolve` and `postresolve` examples only override `dq.appliedPolicy.policyKind` with `pdns.policykinds.NoAction`, it looks like we might be logging the name of a policy that wasn't applied. Perhaps we should check whether `appliedPolicy.d_kind` is different from `DNSFilterEngine::PolicyKind::NoAction` and `wantsRPZ` is true?